packages/core/core/src/requests/BundleGraphRequest.js

```diff
@@ -356,66 +380,19 @@ class BundlerRunner {
       bundleGraphEdgeTypes,
     );
 
-    // Store the serialized bundle graph in an in memory cache so that we avoid serializing it
-    // many times to send to each worker, and in build mode, when writing to cache on shutdown.
-    // Also, pre-compute the hashes for each bundle so they are only computed once and shared between workers.
-    internalBundleGraph.getBundleGraphHash();
-    cacheSerializedObject(internalBundleGraph);
-
-    // Recompute the cache key to account for new dev dependencies and invalidations.
-    let {cacheKey: updatedCacheKey} = await this.getHashes(graph);
     this.api.storeResult(
       {
-        bundlerHash,
         bundleGraph: internalBundleGraph,
         changedAssets: new Map(),
+        assetRequests: [],
       },
-      updatedCacheKey,
+      this.cacheKey,
```

Comment:
Why do we need a cache key here? (And tangentially: what does `storeResult` even do compared to just the return value of the run function?)

Reply:
If you don't call storeResult, nothing is stored in the graph. I forget why... Maybe because in this case the cached result is different from the uncached one (eg changedAssets).

Without a cache key, the result is serialized directly in the request graph, but if a cache key is provided then it is stored as a separate cache entry. We wouldn't want to store the entire bundle graph inside the request graph in this case.